ethcore/src/client/client.rs

```diff
@@ -1318,6 +1318,25 @@ impl BlockChainClient for Client {
 		self.do_virtual_call(&env_info, &mut state, &t, analytics)
 	}
 
+	fn replay_block_transactions(&self, block: BlockId, analytics: CallAnalytics) -> Result<Vec<Executed>, CallError> {
```

Comment:
To share the code with `replay()` you could return a `Result<Box<Iterator<Item = Executed>>, CallError>` here and then `replay()` just becomes:
```
let index = <get-tx-index>;
Ok(self.replay_block(...)?.nth(index))
```